Source/OxyPlot.ImageSharp/PngRenderingContext.cs

```diff
@@ -75,42 +87,286 @@ public void Save(Stream output)
+        {
+            switch (edgeRenderingMode)
+            {
+                case EdgeRenderingMode.PreferGeometricAccuracy:
+                case EdgeRenderingMode.PreferSpeed:
+                    return this.Convert(rect);
+                default:
+                    var actualThickness = this.GetActualThickness(strokeThickness, edgeRenderingMode);
+                    var snapOffset = GetSnapOffset(actualThickness);
+                    return this.ConvertSnap(rect, snapOffset);
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="RectangleF"/>s that should actually be rendered from the list of <see cref="OxyRect"/>s, taking into account DPI scaling and snapping if necessary.
+        /// </summary>
+        /// <param name="rects">The rectangles.</param>
+        /// <param name="strokeThickness">The stroke thickness.</param>
+        /// <param name="edgeRenderingMode">The edge rendering mode.</param>
+        /// <returns>The actual rectangles.</returns>
+        private IEnumerable<RectangleF> GetActualRects(IEnumerable<OxyRect> rects, double strokeThickness, EdgeRenderingMode edgeRenderingMode)
+        {
+            switch (edgeRenderingMode)
+            {
+                case EdgeRenderingMode.PreferGeometricAccuracy:
+                case EdgeRenderingMode.PreferSpeed:
+                    return rects.Select(this.Convert);
+                default:
+                    var actualThickness = this.GetActualThickness(strokeThickness, edgeRenderingMode);
+                    var snapOffset = GetSnapOffset(actualThickness);
+                    return rects.Select(rect => this.ConvertSnap(rect, snapOffset));
+            }
+        }
+
+        /// <summary>
+        /// Gets the stroke thickness that should actually be used for rendering, taking into account DPI scaling and snapping if necessary.
+        /// </summary>
+        /// <param name="strokeThickness">The stroke thickness.</param>
+        /// <param name="edgeRenderingMode">The edge rendering mode.</param>
+        /// <returns>The actual stroke thickness.</returns>
+        private float GetActualThickness(double strokeThickness, EdgeRenderingMode edgeRenderingMode)
+        {
+            var scaledThickness = this.Convert(strokeThickness);
+            if (edgeRenderingMode == EdgeRenderingMode.PreferSharpness)
+            {
+                scaledThickness = Snap(scaledThickness, 0);
+            }
+
+            return scaledThickness;
+        }
+
         /// <inheritdoc/>
         public override void DrawText(ScreenPoint p, string text, OxyColor fill, string fontFamily = null, double fontSize = 10, double fontWeight = 400, double rotation = 0, OxyPlot.HorizontalAlignment horizontalAlignment = OxyPlot.HorizontalAlignment.Left, OxyPlot.VerticalAlignment verticalAlignment = OxyPlot.VerticalAlignment.Top, OxySize? maxSize = null)
```

Comment:
It would be nice to support text rotation, but from a quick search ImageSharp does not seem to support that... Or is there a way?

Reply:
I've done rotated text with ImageSharp before, but it was only mutliples of 90 degrees. I'll dig out that old code and see what I did there - because I feel like I was using transforms in a way that my recent experience suggests don't exist - and get back to you.

Reply:
Nope: that old code is just rotating the whole image. I can't shake the feeling that I must be missing something about ImageSharp, because everything just feels wrong.

As I workaround, we could render the text to another image, rotate that image, and then draw it on top.